lib/scanner/walk.go

```diff
@@ -476,8 +479,11 @@ func SymlinkTypeEqual(disk, index uint32) bool {
 	// know means you are on Unix, and on Unix you don't really care what the
 	// target type is. The moment you do know, and if something doesn't match,
 	// that will propagate through the cluster.
-	if disk&protocol.FlagSymlinkMissingTarget != 0 && index&protocol.FlagSymlinkMissingTarget == 0 {
+	if disk == symlinks.TargetUnknown {
 		return true
 	}
-	return disk&protocol.SymlinkTypeMask == index&protocol.SymlinkTypeMask
+	if f.IsDirectory() {
+		return disk == symlinks.TargetDirectory
+	}
+	return disk == symlinks.TargetFile
```

Comment:
If we know the symlink type (its file), and the other node does not (also ends up being file as we don't check for unknown) we will never announce the fact that we kniw the type.